sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md

```diff
@@ -41,7 +41,13 @@ public final class PushExporter {
   public PushExporter(Collection<MetricProducer> producers) {
     metricExporter = new PushMetricExporter();
     intervalMetricReader =
-        new IntervalMetricReader(producers, metricExporter, 60 /* export interval sec*/);
+        IntervalMetricReader.builder()
+                    .readEnvironment() // Read configuration from environment variables
+                    .readSystemProperties() // Read configuration from system properties
+                    .setExportIntervalMillis(100_000) 
+                    .setMetricExporter(metricExporter)
+                    .setMetricProducers(Collections.singletonList(producers))
```

Comment:
I expect that everything that is not possible to configure via eng/sysprop to be passed as an argument to a factory method `create`

Reply:
This is just a suggestion, please let me know if you think it is better.

Reply:
I'm not clear on how the suggestion would work. Would the `create` method take the builder as a parameter? Would the Builder have a `create` method that took the non-env-configurable parameters, rather than having a `build()` method?

Reply:
I was thinking that Config contains only things that can be configured via env or sysprop everything else is passed as argument to a factory method.

So in this case: `create(Config, List<MetricProducer>, MetricExporter)`

Reply:
What class exposes the `create` method, though?

Reply:
Now I am more confused, because I thought that we do the model of having a Config class as an inner class (see https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java#L313), something like this:

```java
public final class IntervalMetricReader {
 
  public create(Config, List<MetricProducer>, MetricExporter);

  public static final Config {
     public static Config defaultConfig();
     public static Builder newBuilder();

     public static final Builder extends ConfigBuilder<> {
       public Config build();
     }
  }
}
```

Reply:
For classes with many options, e.g. BatchSpanProcessor, I believe a configuration object makes sense. For classes that have a single parameter to configure this becomes a waste of memory and introduces a lot of more code.

Reply:
Only while I was working on the metrics I noticed that having a config object for a single parameter it's a bit overkill and the same result can be archived enhancing the existing builder. 
However, I agree that we have different API for different components:
- Span classes use a config object;
- Metric classes do that directly in the builder.
So, either we use config objects in all classes, or every class uses the standard builder pattern enhanced with support to env/sys vars. 
I am more in favor of the latter since currently, a Config object makes sense IMHO only for the `BatchSpanProcessor`.

Reply:
I am asking for consistency. If we believe builder pattern + support for env/sysprop config in that builder pattern is good then we should go with that model everywhere.

Reply:
I would argue to use the build pattern + support of env/sys. My motivation is that this way, the API surface is not increased and memory is not used to store the `Configuration` class. I have changed the PR message so it does not resolve the issue and I will adapt the span processors to use only the builder.

Reply:
Sure thing, @trask does that work for you in auto instrumentation?

Reply:
Yes, this is looking good @thisthat 👍. Thanks for including `readProperties(Properties)`, we will probably use that for something like an `opentelemetry-auto.properties` file (in addition to calling `readEnvironment` and `readSystemProperties`).

Only (very) minor thought is maybe rename `readEnvironment` to `readEnvironmentVariables`.

Reply:
Oh, I forgot, what about endpoint configuration for the otlp exporters? Are we waiting for https://github.com/open-telemetry/opentelemetry-specification/issues/172 to be resolved before adding? Just curious. Thanks.

Reply:
I was waiting to resolve #996 and then report in https://github.com/open-telemetry/opentelemetry-specification/issues/572 what the Java current implementation uses for the variable names. I don't think https://github.com/open-telemetry/opentelemetry-specification/issues/172 is blocking us since it is a matter of renaming of few constants and updating the documentation in the worst case

Reply:
Filed https://github.com/open-telemetry/opentelemetry-java/issues/1194 to not forget about the suggestion from @trask